### PR TITLE
Deprecate `center` on `df.expanding`

### LINKED ISSUE #20647: API/BUG: center=True in expanding operations returns non-sensical results
What is `center=True` supposed to do in `df.expanding()...` operations?

Using the example from the docstring of `.expanding()`:

```
In [103]: df = DataFrame({'B': [0, 1, 2, np.nan, 4]})

In [105]: df.expanding(2).sum()
Out[105]: 
     B
0  NaN
1  1.0
2  3.0
3  3.0
4  7.0
```

Now adding `center=True`:

```
In [106]: df.expanding(2, center=True).sum()
Out[106]: 
     B
0  3.0
1  3.0
2  7.0
3  7.0
4  6.0
```

Two observations: it seems it just shifted the first rows (which is not the same as centering) and I don't know where the last values are coming from (and they should certainly not decrease again in an expanding sum with only positive values). 

Further, I also cannot think of what it actually *should* do. If we really want to center it, that would mean returning with a new index (as the real center would only increase with 0.5 and be something like [0, 0.5, 1, 1.5, 2] for those data).

So maybe we should rather remove this keyword for `expanding` ?

### PR DESCRIPTION
**Pull request by MBrouns**
`df.expanding(center=True)` currently returns non-sensical results and it is unclear what results would be expected. It was previously removed in #7934 for that same reason,but was reintroduced for unclear reasons in a later refactoring

- [x] closes #20647
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/window/expanding.py [hunk 1]
```diff
@@ -58,6 +59,16 @@ class Expanding(_Rolling_and_Expanding):
     _attributes = ["min_periods", "center", "axis"]
 
     def __init__(self, obj, min_periods=1, center=False, axis=0, **kwargs):
+        if center:
```
Comment (charlesdong1991) [inline pandas/core/window/expanding.py:62]: will this `center` kwarg be deprecated entirely? so users shouldn't use `center` no matter if this is set to `True` or `False`, and now using `if center` won't raise warning if setting to `False` which i _think_ do not align with this deprecation purpose.

**Review by charlesdong1991 — COMMENTED** [review]
thanks for active PRs! not very familiar with this deprecation scope, have a question to clear it.

File: pandas/core/window/expanding.py (hunk 1 above)
Comment (MBrouns) [reply pandas/core/window/expanding.py:62]: That's a good point, basically the warning needs to be raised whenever center is set explicitly.

File: doc/source/whatsnew/v1.1.0.rst [hunk 2]
```diff
@@ -781,6 +781,7 @@ Deprecations
 - The ``squeeze`` keyword in the ``groupby`` function is deprecated and will be removed in a future version (:issue:`32380`)
 - The ``tz`` keyword in :meth:`Period.to_timestamp` is deprecated and will be removed in a future version; use `per.to_timestamp(...).tz_localize(tz)`` instead (:issue:`34522`)
 - :meth:`DatetimeIndex.to_perioddelta` is deprecated and will be removed in a future version.  Use ``index - index.to_period(freq).to_timestamp()`` instead (:issue:`34853`)
+- the ``center`` keyword in the :meth:`DataFrame.expanding`  function is deprecated and will be removed in a future version (:issue:`20647`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:784]: capital here

File: pandas/core/window/expanding.py [hunk 3]
```diff
@@ -57,7 +58,19 @@ class Expanding(_Rolling_and_Expanding):
 
     _attributes = ["min_periods", "center", "axis"]
 
-    def __init__(self, obj, min_periods=1, center=False, axis=0, **kwargs):
+    def __init__(self, obj, min_periods=1, center=None, axis=0, **kwargs):
+        if center is not None:
+            warnings.warn(
+                """
+                the `center` argument on `expanding` will be removed in the future.
+                Please be aware that there are issues with its current implementation,
```
Comment (jreback) [inline pandas/core/window/expanding.py:66]: you only need the 1st sentence

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
you will either have to explicity set `center=None` for expanding tests, or use a `@pytest.mark.filterwarnings("...")` for this specific warning (this is ok to do)

File: pandas/core/window/expanding.py [hunk 4]
```diff
@@ -57,7 +58,15 @@ class Expanding(_Rolling_and_Expanding):
...
-    def __init__(self, obj, min_periods=1, center=False, axis=0, **kwargs):
+    def __init__(self, obj, min_periods=1, center=None, axis=0, **kwargs):
+        if center is not None:
+            warnings.warn(
+                "The `center` argument on `expanding` will be removed in the future",
+                FutureWarning,
+                stacklevel=3,
+            )
+        else:
+            center = False
```
Comment (charlesdong1991) [inline pandas/core/window/expanding.py:69]: will this change break current tests when `center=True`

File: pandas/core/window/expanding.py [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import warnings
```
Comment (charlesdong1991) [inline pandas/core/window/expanding.py:1]: might need to sort imports using `isort` to fix CI failure

File: pandas/core/window/expanding.py (hunk 4 above)
Comment (MBrouns) [reply pandas/core/window/expanding.py:69]: It shouldn't. In that case the first branch will hit, leaving `center` unaltered and only throwing the warning

Comment (MBrouns) [reply pandas/core/window/expanding.py:69]: It shouldn't. In that case `center is not None` is true, meaning center will be kept unaltered.

**Comment by MBrouns** [comment]
@jreback If I understood correctly you mean the change I made in https://github.com/pandas-dev/pandas/pull/34887/commits/f47dedebac6b43c1678ba183305277423e5a24d9

**Comment by jreback** [comment]
> @jreback If I understood correctly you mean the change I made in [f47dede](https://github.com/pandas-dev/pandas/commit/f47dedebac6b43c1678ba183305277423e5a24d9)

yep

**Comment by MBrouns** [comment]
@jreback anything that still needs to happen here?

**Comment by charlesdong1991** [comment]
@MBrouns  pls resolve conflicts and rebase, and make CI green. otherwise, it's unlikely that @jreback will review this PR

**Comment by MBrouns** [comment]
@charlesdong1991 the test failures seem to be unrelated to this PR

**Comment by charlesdong1991** [comment]
@MBrouns  hey, after a quick skim, i think the linting failure is indeed on master, which will be fixed by #35007 

but i think for this PR, still need to make changes for the test in `pandas/tests/window/test_expanding.py` in which the test uses `expanding` and raise the warning. 

you need to either catch or ignore the warnings for the test, and the test seems to be `test_expanding_count_with_min_periods`.

File: pandas/core/base.py [hunk 6]
```diff
@@ -581,12 +582,17 @@ def _shallow_copy(self, obj, **kwargs):
         """
         return a new object with the replacement attributes
         """
-        if isinstance(obj, self._constructor):
-            obj = obj.obj
-        for attr in self._attributes:
-            if attr not in kwargs:
-                kwargs[attr] = getattr(self, attr)
-        return self._constructor(obj, **kwargs)
+        with warnings.catch_warnings():
```
Comment (WillAyd) [inline pandas/core/base.py:585]: Do you know where this is getting called? Seems strange to have to filter the warning here

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/base.py (hunk 6 above)
Comment (jreback) [reply pandas/core/base.py:585]: oh no, we DO not want to do this here at all

File: pandas/core/generic.py [hunk 7]
```diff
@@ -10501,7 +10501,7 @@ def rolling(
         cls.rolling = rolling
 
         @doc(Expanding)
-        def expanding(self, min_periods=1, center=False, axis=0):
+        def expanding(self, min_periods=1, center=None, axis=0):
```
Comment (jreback) [inline pandas/core/generic.py:10504]: you can  warn right here, and if its None, don't pass it thru, just set it to False

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/window/expanding.py [hunk 8]
```diff
@@ -129,7 +130,12 @@ def aggregate(self, func, *args, **kwargs):
     @Substitution(name="expanding")
     @Appender(_shared_docs["count"])
     def count(self, **kwargs):
-        return super().count(**kwargs)
+        with warnings.catch_warnings():
```
Comment (jreback) [inline pandas/core/window/expanding.py:133]: this should not show a warning at all

File: pandas/tests/window/test_expanding.py [hunk 9]
```diff
@@ -18,24 +18,17 @@ def test_doc_string():
...
-    c(min_periods=1, center=True)
-    c(min_periods=1, center=False)
 
     # not valid
     for w in [2.0, "foo", np.array([2])]:
         msg = "min_periods must be an integer"
         with pytest.raises(ValueError, match=msg):
             c(min_periods=w)
 
-        msg = "center must be a boolean"
```
Comment (jreback) [inline pandas/tests/window/test_expanding.py:35]: you can leave this one (you can filter the test warning  like

```@pytest.mark.filterwarnings("ignore:elementwise comparison failed:FutureWarning")```

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @MBrouns very nice

**Pull request merged**